realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1051,7 +1051,11 @@ std::unique_ptr<Query> handoverQueryToWorker(jlong bgSharedGroupPtr, jlong query
     if (sg->get_transact_stage() != SharedGroup::transact_Reading) {
         // if the SharedGroup is not in Read Transaction, we position it at the same version as the handover
         sg->begin_read(handoverQuery->version);
-    } else if (sg->get_version_of_current_transaction() != handoverQuery->version) {
+    } else if (sg->get_version_of_current_transaction() < handoverQuery->version) {
+        // going forward
+        LangBindHelper::advance_read(*sg, handoverQuery->version);
+    } else if (handoverQuery->version < sg->get_version_of_current_transaction()) {
+        // going backward
```

Comment:
I'll need to some time to think through the flow of this, especially regarding where the SharedGroup is actually created in the flow.